### PR TITLE
[TypeScript][Skill] Fix undefined state issue

### PR DESCRIPTION
**Pull request by LCarbonellRod**
### Purpose
*What is the context of this pull request? Why is it being done?*

### Changes
*Are there any changes that need to be called out as significant or particularly difficult to grasp? (Include illustrative screenshots for context if applicable.)*

### Tests
*Is this covered by existing tests or new ones? If no, why not?*

### Feature Plan
*Are there any remaining steps or dependencies before this issue can be fully resolved? If so, describe and link to any relevant pull requests or issues.*

#### General
- [ ] I have commented my code, particularly in hard-to-understand areas	
- [ ] I have added or updated the appropriate tests	
- [ ] I have updated related documentation

### CODE REVIEW

File: templates/typescript/samples/sample-skill/src/dialogs/sampleDialog.ts [hunk 1]
```diff
@@ -38,8 +38,8 @@ export class SampleDialog extends SkillDialogBase {
 
         const sample: ((sc: WaterfallStepContext) => Promise<DialogTurnResult>)[] = [
             // NOTE: Uncomment these lines to include authentication steps to this dialog
-            // GetAuthToken,
-            // AfterGetAuthToken,
+            this.getAuthToken.bind(this),
+            this.afterGetAuthToken.bind(this),
```
Comment (Batta32) [inline templates/typescript/samples/sample-skill/src/dialogs/sampleDialog.ts:42]: These changes won't be in 1.0 right? This is something to report to tomSauret

File: templates/typescript/samples/sample-skill/src/dialogs/skillDialogBase.ts [hunk 2]
```diff
@@ -46,12 +47,17 @@ export class SkillDialogBase extends ComponentDialog {
...
+        {
+            throw new Error("You must configure an authentication connection in your bot file before using this component.");
+        }
+        const oAuthSettings: OAuthPromptSettings[] = 
+        [{
+            connectionName: 'Outlook',
+            title: 'TitleText',
+            text: 'TextText'
+        }];
+        this.addDialog(new MultiProviderAuthDialog(settings.oauthConnections, oAuthSettings));
```
Comment (Batta32) [inline templates/typescript/samples/sample-skill/src/dialogs/skillDialogBase.ts:60]: Same comment of above

File: templates/typescript/samples/sample-skill/src/dialogs/skillDialogBase.ts [hunk 3]
```diff
@@ -75,7 +81,7 @@ export class SkillDialogBase extends ComponentDialog {
 
             if (providerTokenResponse !== undefined) {
                 // eslint-disable-next-line @typescript-eslint/no-explicit-any
-                const state: any = await this.stateAccessor.get(sc.context);
+                const state: any = await this.stateAccessor.get(sc.context, { token: '', timeZone: new Date(), clear: () => { return; }});
```
Comment (Batta32) [inline templates/typescript/samples/sample-skill/src/dialogs/skillDialogBase.ts:84]: This is something to be added in 1.0 of Skill

File: templates/typescript/samples/sample-skill/src/index.ts [hunk 4]
```diff
@@ -83,7 +84,7 @@ const cosmosDbStorageOptions: CosmosDbPartitionedStorageOptions = {
 const storage: CosmosDbPartitionedStorage =  new CosmosDbPartitionedStorage(cosmosDbStorageOptions);
 const userState: UserState = new UserState(storage);
 const conversationState: ConversationState = new ConversationState(storage);
-const stateAccessor: StatePropertyAccessor<SkillState> = userState.createProperty(SkillState.name);
+const stateAccessor: StatePropertyAccessor<SkillState> = conversationState.createProperty(SkillState.name);
```
Comment (Batta32) [inline templates/typescript/samples/sample-skill/src/index.ts:87]: This is something to be added in 1.0 of Skill

**Comment by LCarbonellRod** [comment]
> Some comments to clarify these changes, I'll try to review the changes to initialize the `SkillState`

All the changes you are pointing out are needed to the Skill works successfully. If I'll give this branch to the user, this changes should be here, don't they?

**Comment by Batta32** [comment]
Just to confirm that you won't add these changes to the 1.0, but only the initialization of the `SkillState`

**Comment by LCarbonellRod** [comment]
No, this PR is not for 1.0 changes but only to user use right?

**Comment by Batta32** [comment]
These changes should be in 1.0:
* https://github.com/southworks/botframework-solutions/pull/419/files#diff-778fd5fad5e7b265da367245c4ce8193R84
* https://github.com/southworks/botframework-solutions/pull/419/files#diff-96a4f8f9c84291a95ce5b7235e19bb65L86-R87

The rest, don't need to be in 1.0.

Another things, is it okay to be pointed to next? The branch that we already sent to the user was from master I guess

**Comment by LCarbonellRod** [comment]
> These changes should be in 1.0:
> 
> * https://github.com/southworks/botframework-solutions/pull/419/files#diff-778fd5fad5e7b265da367245c4ce8193R84
> * https://github.com/southworks/botframework-solutions/pull/419/files#diff-96a4f8f9c84291a95ce5b7235e19bb65L86-R87
> 
> The rest, don't need to be in 1.0.
> 
> Another things, is it okay to be pointed to next? The branch that we already sent to the user was from master I guess

Yes, you are right, I will create another PR with the changes pointing to master and close this if there are no problem.

**Pull request closed without merging**